### PR TITLE
#360 - Create stress/load test which should cover all APIs with mocked OP

### PR DESCRIPTION
**Pull request by duttarnab**
#360 - Create stress/load test which should cover all APIs with mocked OP
https://github.com/GluuFederation/oxd/issues/360

We have written mocks for `RegisterClient` and `TokenClient`. and running test cases for `RegisterSiteOperation` and `GetClientTokenOperation`

Could you please verify if we are going in right direction. If yes then we can implement same for all the clients.

### CODE REVIEW

File: oxd-server/src/main/java/org/gluu/oxd/server/Processor.java [hunk 1]
```diff
@@ -36,10 +37,10 @@ public Processor(ValidationService validationService) {
         this.validationService = validationService;
     }
 
-    public IOpResponse process(Command command) {
+    public IOpResponse process(Command command, OpClientFactory opClientFactory) {
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/Processor.java]: remove it, it should be taken from guice injector.

File: oxd-server/src/main/java/org/gluu/oxd/server/RestResource.java [hunk 2]
```diff
@@ -21,7 +23,12 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(RestResource.class);
 
-    public RestResource() {
+
+    private static OpClientFactory opClientFactory;
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/RestResource.java]: Redundant.

File: oxd-server/src/main/java/org/gluu/oxd/server/guice/GuiceModule.java [hunk 3]
```diff
@@ -34,5 +38,6 @@ protected void configure() {
         bind(DiscoveryService.class).in(Singleton.class);
         bind(ValidationService.class).in(Singleton.class);
         bind(StateService.class).in(Singleton.class);
+        bind(OpClientFactory.class).to(OpClientFactoryImpl.class).in(Singleton.class);
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/guice/GuiceModule.java]: This is correct, now we need same but in `TestGuiceModule` which bind to mock factory.
```
bind(OpClientFactory.class).to(MockOpClientFactory.class).in(Singleton.class);
```

File: oxd-server/src/main/java/org/gluu/oxd/server/op/GetClientTokenOperation.java [hunk 4]
```diff
@@ -31,21 +31,24 @@
...
+    private OpClientFactory opClientFactory;
     /**
      * Base constructor
      *
      * @param command command
      */
-    protected GetClientTokenOperation(Command command, final Injector injector) {
+    protected GetClientTokenOperation(Command command, final Injector injector, OpClientFactory opClientFactory) {
         super(command, injector, GetClientTokenParams.class);
+        this.opClientFactory = opClientFactory;
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/op/GetClientTokenOperation.java]: All these references are redundant, add to `BaseOperation`, it's responsibility of guice to construct it.
```
public OpClientFactory getOpClientFactory () {
        return getInstance(OpClientFactory .class);
    }
```

File: oxd-server/src/main/java/org/gluu/oxd/server/op/GetClientTokenOperation.java [hunk 5]
```diff
@@ -31,21 +31,24 @@
...
     }
 
     @Override
     public IOpResponse execute(GetClientTokenParams params) {
         try {
             final AuthenticationMethod authenticationMethod = AuthenticationMethod.fromString(params.getAuthenticationMethod());
             final String tokenEndpoint = getDiscoveryService().getConnectDiscoveryResponse(params.getOpHost(), params.getOpDiscoveryPath()).getTokenEndpoint();
-            final TokenClient tokenClient = new TokenClient(tokenEndpoint);
+            //final TokenClient tokenClient = new TokenClient(tokenEndpoint);
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/op/GetClientTokenOperation.java]: redundant line

File: oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+/**
+ * Main class to set up and tear down suite.
+ *
+ * @author Yuriy Zabrovarnyy
+ * @version 0.9, 21/08/2013
+ */
+
```
Comment (yuriyz) [inline oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java]: Run it with Test Guice Module
```
@Guice(modules = org.gluu.oxd.mock.MockAppModule.class)
```

File: oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            setupSwaggerSuite(MockTester.getTargetHost(host), opHost, redirectUrls);
+            LOG.debug("Finished beforeSuite!");
+        } catch (Exception e) {
+            LOG.error("Failed to start suite.", e);
+            throw new AssertionError("Failed to start suite.");
+        }
+    }
+
+    private static void setOpClientFactory() {
```
Comment (yuriyz) [inline oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java]: redundant file

File: oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    private static final Logger LOG = LoggerFactory.getLogger(SetUpTest.class);
+
+    public static DropwizardTestSupport<OxdServerConfiguration> SUPPORT = null;
+
+
+    @Parameters({"host", "opHost", "redirectUrls"})
+    @BeforeSuite
+    public static void beforeSuite(String host, String opHost, String redirectUrls) {
+        try {
```
Comment (yuriyz) [inline oxd-server/src/test/java/org/gluu/oxd/server/mock/op/MockSetUpTest.java]: All this is copy/paste from `org.gluu.oxd.server.SetUpTest#beforeSuite`. Lets refactor existing code and avoid it.

**Comment by duttarnab** [comment]
Raised another PR #365 addressing this ticket.

**Pull request closed without merging**